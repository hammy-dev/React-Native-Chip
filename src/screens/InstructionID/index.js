import { useState } from 'react';
import {
  View, Text, TouchableOpacity, Image, Alert,
} from 'react-native';
import * as ImagePicker from 'expo-image-picker';
import Modal from 'react-native-modal';
import Chipbutton from '../../components/buttons/index';
import Circle from '../../components/Circles/index';
import styles from './style';
import SignUp from '../../assets/SignUp/signup4.png';
import homeIcon from '../../assets/homeicon/ic_close_green.png';
import camera from '../../assets/camera.png';
import gallery from '../../assets/gallery.png';

function InstructionID({ navigation, route }) {
  const [visibility, setVisible] = useState(false);
  const data = route.params;
  const pickImage = async () => {
    // No permissions request is necessary for launching the image library
    const result = await ImagePicker.launchImageLibraryAsync({
      mediaTypes: ImagePicker.MediaTypeOptions.All,
      allowsEditing: true,
      aspect: [4, 3],
      quality: 1,
    });
    if (!result.cancelled && result.uri) {
      data.identity_img = result.uri;
      navigation.navigate('UploadDoc', { ...data });
    }
  };
  const openCamera = async () => {
    const permissionResult = await ImagePicker.requestCameraPermissionsAsync();

    if (permissionResult.granted === false) {
      Alert.alert("You've refused to allow this appp to access your camera!");
      return;
    }

    const result = await ImagePicker.launchCameraAsync();

    if (!result.cancelled) {
      const localUri = result.uri;
      if (localUri) {
        data.identity_img = localUri;
        navigation.navigate('UploadDoc', { ...data });
      }
    }
  };

  return (
    <View style={styles.screenContainer}>
      <View style={styles.circleContainer}>
        <Circle textStyle={styles.circletext} text="1" style={styles.circleOne} />
        <View style={styles.lineOne} />
        <Circle textStyle={styles.title} text="2" style={styles.circleTwo} />
        <View style={styles.lineTwo} />
        <Circle textStyle={styles.title} text="3" style={styles.circleThree} />
      </View>
      <View style={styles.circleTextContainer}>
        <Text style={styles.circleTextOne}>Upload Identity</Text>
        <Text style={styles.circleTextTwo}>Upload document</Text>
        <Text style={styles.circleTextThree}>Sign Up</Text>
      </View>
      <View style={styles.cardMainContainer}>
        <View style={styles.cardContainer}>
          <View>
            <Circle style={styles.insideCardCircle} img={SignUp} />
          </View>
          <View style={styles.insideCardTextContainerOne}>
            <Text style={styles.insideCardTextOne}>
              Upload a picture of the directors ID/
              {'\n'}
              {' '}
              Driver License or Passport
            </Text>
          </View>

          <View style={styles.insideCardChipButtonContainer}>
            <Chipbutton
              text="Upload"
              testID="button"
              customStyle={styles.insideCardChipButton}
              customText={styles.insideChipButtonText}
              // onPress={() => { pickImage(); }}
              onPress={() => { setVisible(true); }}
              activeOpacity={0.7}
            />
          </View>
        </View>
      </View>
      <Modal visible={visibility} style={styles.modal}>
        <View style={styles.modalView}>
          <TouchableOpacity onPress={() => { setVisible(false); }} style={styles.modalClose} testID="close">
            <Image source={homeIcon} style={styles.home} />
          </TouchableOpacity>
          <View><Text style={styles.chooseImg}>Choose Image</Text></View>
          <View style={styles.modalContainer}>
            <TouchableOpacity activeOpacity={0.7} testID="camera" onPress={() => { openCamera(); }}>
              <View style={styles.cameraView}>
                <Image source={camera} style={styles.cameraImg} />
              </View>
              <Text style={styles.cameraText}>Camera</Text>
            </TouchableOpacity>
            <TouchableOpacity activeOpacity={0.7} testID="pickImage" onPress={() => { pickImage(); }}>
              <View style={styles.galleryView}>
                <Image source={gallery} style={styles.galleryImg} />
              </View>
              <Text style={styles.galleryText}>Gallery</Text>
            </TouchableOpacity>
          </View>
        </View>
      </Modal>
    </View>
  );
}

export default InstructionID;
