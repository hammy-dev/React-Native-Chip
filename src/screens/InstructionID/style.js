import {
  StyleSheet,
} from 'react-native';
import Colors from '../../constants/Colors';

export default StyleSheet.create({
  screenContainer: { flex: 1, backgroundColor: Colors.white },
  circleContainer: {
    flex: 0.2, backgroundColor: Colors.white, flexDirection: 'row', alignItems: 'center', justifyContent: 'center', marginTop: '40%',
  },
  circleOne: {
    backgroundColor: Colors.uploadDocsCircleText, alignSelf: 'center', borderColor: Colors.uploadDocsCircleText, borderWidth: 1,
  },
  circleTwo: {
    backgroundColor: Colors.white, alignSelf: 'center', borderColor: Colors.grey, borderWidth: 1,
  },
  circleThree: {
    backgroundColor: Colors.white, alignSelf: 'center', borderColor: Colors.grey, borderWidth: 1,
  },
  lineOne: {
    alignSelf: 'center', width: '20%', height: 1, alignItems: 'center', backgroundColor: Colors.UploadDocsCircleThree,
  },
  lineTwo: {
    alignSelf: 'center', width: '20%', height: 1, alignItems: 'center', backgroundColor: Colors.UploadDocsCircleThree,
  },
  circleTextContainer: {
    flex: 0.1, backgroundColor: Colors.white, flexDirection: 'row', alignItems: 'flex-start', justifyContent: 'space-around', marginTop: '-3%',
  },
  circleTextOne: { fontSize: 15, color: Colors.circle },
  circleTextTwo: { fontSize: 15, marginRight: '2%', color: Colors.darkgrey },
  circleTextThree: { fontSize: 15, color: Colors.darkgrey },
  cardMainContainer: {
    flex: 0.9, backgroundColor: Colors.white, flexDirection: 'row', alignItems: 'center', justifyContent: 'center',
  },
  cardContainer: {
    width: '80%', height: '70%', backgroundColor: Colors.uploadDocsCardtext, borderWidth: 1, borderRadius: 15, borderColor: Colors.uploadDocsCardtext,
  },
  insideCardCircle: {
    backgroundColor: Colors.white, padding: 30, alignSelf: 'center', overflow: 'hidden', width: 170, height: 170, borderRadius: 170 / 2, marginTop: '-25%',
  },
  insideCardTextContainerOne: {
    flex: 0.3, backgroundColor: Colors.uploadDocsCardtext, marginTop: '9%', flexDirection: 'row', alignItems: 'center', justifyContent: 'center',
  },
  insideCardTextOne: { fontSize: 17, textAlign: 'center' },
  insideCardTextContainerTwo: {
    flex: 0.2, backgroundColor: Colors.uploadDocsCardtext, flexDirection: 'row', alignItems: 'center', justifyContent: 'center', marginTop: '-3%',
  },
  insideCardTextTwo: { fontSize: 17 },
  insideCardChipButtonContainer: {
    flex: 0.5, backgroundColor: Colors.uploadDocsCardtext, flexDirection: 'row', alignItems: 'center', justifyContent: 'center',
  },
  insideCardChipButton: {
    backgroundColor: Colors.InstructionButton, width: 180, height: 60, justifyContent: 'center',
  },
  insideChipButtonText: { fontSize: 20, color: Colors.white },
  title: { color: Colors.grey, fontSize: 20 },
  modal: { margin: 0, backgroundColor: Colors.transparent },
  modalView: {
    height: '30%',
    marginTop: 'auto',
    backgroundColor: Colors.white,
    borderTopLeftRadius: 20,
    borderTopRightRadius: 20,
    overflow: 'hidden',
  },
  modalClose: {
    alignSelf: 'flex-end',
    marginRight: 10,
    marginTop: 10,
  },
  home: { width: 30, height: 30 },
  chooseImg: { fontSize: 20, alignSelf: 'center' },
  circletext: { color: Colors.white, fontSize: 20 },
  modalContainer: { flexDirection: 'row', justifyContent: 'space-evenly', marginTop: 20 },
  cameraView: { overflow: 'hidden', width: 100, height: 100 },
  cameraImg: { width: '100%', height: '100%' },
  cameraText: { alignSelf: 'center' },
  galleryView: { overflow: 'hidden', width: 90, height: 90 },
  galleryImg: { width: '100%', height: '100%' },
  galleryText: { alignSelf: 'center', marginTop: 11 },
});
