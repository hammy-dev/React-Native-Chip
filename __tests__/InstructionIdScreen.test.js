import React from 'react';
import renderer from 'react-test-renderer';
import InstructionID from '../src/screens/InstructionID/index';
import {  render, fireEvent } from '@testing-library/react-native';
import * as ImagePicker from 'expo-image-picker';


beforeAll(() => {
  ImagePicker.launchImageLibraryAsync= jest.fn().mockResolvedValue({uri:"55555",cancelled :false});
  ImagePicker.requestCameraPermissionsAsync= jest.fn().mockResolvedValue({uri:"55555",cancelled :false, granted :true});
  ImagePicker.launchCameraAsync= jest.fn().mockResolvedValue({uri:"55555",cancelled :false});
});

describe('InstructionId', () => {
    it('renders correctly',  async() => {
      const mockedParams = {
        route: { params: { currentBid: 'whatever-id' } },
        navigation: {navigate:jest.fn()}
      };
      
      
        const tree = renderer.create(<InstructionID {...mockedParams} />).toJSON();
        expect(tree.children.length).toBe(4);
      
       
        global.fetch = jest.fn().mockImplementation(() => mockFetch);
        const component= render(<InstructionID {...mockedParams} />);
        const nextPress = component.getByTestId('button');
        
        fireEvent.press(nextPress);
        expect(nextPress).toBeTruthy();

        const close = component.getByTestId('close');
        fireEvent.press(close);
        expect(close).toBeTruthy();

        const camera = component.getByTestId('camera');
        fireEvent.press(camera);
        expect(camera).toBeTruthy();

        const pickImage = component.getByTestId('pickImage');
        fireEvent.press(pickImage);
        expect(pickImage).toBeTruthy();

       
    });
  });