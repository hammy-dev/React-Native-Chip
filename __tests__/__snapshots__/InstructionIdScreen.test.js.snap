// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstructionId renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "flex": 0.2,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginTop": "40%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#13bf62",
          "borderColor": "#13bf62",
          "borderRadius": 30,
          "borderWidth": 1,
          "height": 60,
          "justifyContent": "center",
          "textAlign": "center",
          "width": 60,
        }
      }
    >
      <Text
        style={Object {}}
      >
        1
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#cdcdcd",
          "height": 1,
          "width": "20%",
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "white",
          "borderColor": "gray",
          "borderRadius": 30,
          "borderWidth": 1,
          "height": 60,
          "justifyContent": "center",
          "textAlign": "center",
          "width": 60,
        }
      }
    >
      <Text
        style={Object {}}
      >
        2
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#cdcdcd",
          "height": 1,
          "width": "20%",
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "white",
          "borderColor": "gray",
          "borderRadius": 30,
          "borderWidth": 1,
          "height": 60,
          "justifyContent": "center",
          "textAlign": "center",
          "width": 60,
        }
      }
    >
      <Text
        style={Object {}}
      >
        3
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "backgroundColor": "white",
        "flex": 0.1,
        "flexDirection": "row",
        "justifyContent": "space-around",
        "marginTop": "-3%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#00c357",
          "fontSize": 15,
        }
      }
    >
      Instruction
    </Text>
    <Text
      style={
        Object {
          "color": "#aaaaaa",
          "fontSize": 15,
          "marginRight": "2%",
        }
      }
    >
      Upload document
    </Text>
    <Text
      style={
        Object {
          "color": "#aaaaaa",
          "fontSize": 15,
        }
      }
    >
      Sign Up
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "flex": 0.9,
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#e3f1f4",
          "borderColor": "#e3f1f4",
          "borderRadius": 15,
          "borderWidth": 1,
          "height": "70%",
          "width": "80%",
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "backgroundColor": "white",
              "borderColor": "white",
              "borderRadius": 85,
              "borderWidth": 1,
              "height": 170,
              "justifyContent": "center",
              "marginTop": "-25%",
              "textAlign": "center",
              "width": 170,
            }
          }
        >
          <Text
            style={Object {}}
          />
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#e3f1f4",
            "flex": 0.3,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginTop": "9%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 17,
              "textAlign": "center",
            }
          }
        >
          Upload a picture of the director's ID/ 
          

           Driver License or Passport
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#e3f1f4",
            "flex": 0.5,
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#0d7281",
              "borderRadius": 30,
              "height": "50%",
              "opacity": 1,
              "width": "48%",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "white",
                  "fontSize": 18,
                },
                Object {
                  "color": "white",
                  "fontSize": 20,
                  "paddingTop": 10,
                },
              ]
            }
          >
            Upload
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
